fgd_cc_contacts/fgd_cc_contacts.py

```diff
@@ -0,0 +1,167 @@
+
+    # Load coded demog surveys
+    with open(demog_surveys_input_path, "r") as f:
+        surveys = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
+
+    # Filter out people who haven't answered the fgd_cc consent question
+    fgd_cc_consent_key = "Response_1 (Category) - wt_fgd_cc"
+    fgd_cc_data = [td for td in fgd_cc_data if fgd_cc_consent_key in td]
+
+    # Apply the demog surveys to the fgd_cc data
+    TracedData.update_iterable(user, "avf_phone_id", fgd_cc_data, surveys, "surveys")
+
+    # Annotate fgd_cc_data with whether or not the respondent's is from Mogadishu
+    mogadishu_districts = [
+        "mogadishu",
+        "mogadisho",  # TODO: Remove need for this by correcting Coda file
+        "boondheere",
+        "cabdiasis",
+        "daynile",
+        "dharkenley",
+        "heliwa",
+        "hodan",
+        "hawl wadaag",
+        "karaan",
+        "shangaani",
+        "shibis",
+        "waaberi",
+        "wadajir",
+        "wardhiigleey",
+        "xamar jajab",
+        "xamar weyn",
+        "yaaqshid"
+    ]
+    raw_district_key = "District (Text) - wt_demog_1"
+    coded_district_key = "District (Text) - wt_demog_1_coded"
+    raw_age_key = "Age (Text) - wt_demog_2"
+    coded_age_key = "Age (Text) - wt_demog_2_coded"
+    raw_gender_key = "Gender (Text) - wt_demog_1"
+    coded_gender_key = "Gender (Text) - wt_demog_1_coded"
+    for td in fgd_cc_data:
+        if raw_district_key not in td:
+            td.append_data({
+                raw_district_key: "NA"
+            }, Metadata(user, Metadata.get_call_location(), time.time()))
+
+        if coded_district_key not in td:
+            td.append_data({
+                coded_district_key: "none"
+            }, Metadata(user, Metadata.get_call_location(), time.time()))
+
+        is_mogadishu = td[coded_district_key] in mogadishu_districts
+        td.append_data({
+            "Raw FGD/CC Response": td["Response_1 (Text) - wt_fgd_cc"],
+            "FGD/CC Consented": td[fgd_cc_consent_key],
+            "Raw District": td[raw_district_key],
+            "Coded District":
+                td[coded_district_key] if td[raw_district_key] != Codes.TRUE_MISSING else Codes.TRUE_MISSING,
+            "Coded Age": td[coded_age_key] if td[raw_age_key] != Codes.TRUE_MISSING else Codes.TRUE_MISSING,
+            "Coded Gender": td[coded_gender_key] if td[raw_gender_key] != Codes.TRUE_MISSING else Codes.TRUE_MISSING,
+            "District Mogadishu": "Yes" if is_mogadishu else "No",
+            "FGD/CC Consented and District Mogadishu":
+                "Yes" if is_mogadishu and td[fgd_cc_consent_key] == "Yes" else "No"
+        }, Metadata(user, Metadata.get_call_location(), time.time()))
+
+    # Filter out respondents who aren't from Mogadishu
+    fgd_cc_data = [td for td in fgd_cc_data if td["District Mogadishu"] == "Yes"]
+
+    # Filter out respondents who are definitely younger than 18
```

Comment:
Given this aim, consider copying all the data and then walking over it to remote items where age < 18.

The pattern of

```
try:
  do (x)
except:
  do (x)
```
has a few concerns.